src/store/messages/saga.ts

```diff
@@ -243,7 +244,32 @@ export function* createOptimisticMessage(channelId, message, parentMessage, file
   const existingMessages = yield select(rawMessagesSelector(channelId));
   const currentUser = yield select(currentUserSelector);
 
-  const temporaryMessage = createOptimisticMessageObject(message, currentUser, parentMessage, file, rootMessageId);
+  let enrichedFile = file;
+  if (
+    file &&
+    file.mediaType === MediaType.Image &&
+    file.nativeFile &&
+    (file.width === undefined || file.height === undefined)
+  ) {
+    try {
+      const { width, height } = yield call(getImageDimensions, file.nativeFile);
+      enrichedFile = { ...file, width, height };
```

Comment:
We essentially copy the dimensions of the actual image in the optimistic message object, that the placeholder image has the same dimensions as the actual image, which prevents it from re-sizing after the file is uploaded